zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -563,6 +569,12 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/UserTaskSearchQueryResponse"
+            application/vnd.camunda.api.keys.number+json:
+              schema:
+                $ref: "#/components/schemas/UserTaskSearchQueryResponse"
+            application/vnd.camunda.api.keys.string+json:
+              schema:
+                $ref: "#/components/schemas/UserTaskSearchQueryResult"
```

Comment:
💭 If I'm understanding correctly, we are distinguishing between the two types of keys by naming schemas as a "Result" or a "Response".

Is there a reason to not be more explicit about those different types, like including "number" or "string" in the name? It seems likely that (a) someone new to the project would take longer to intuit why some schemas were named Result and some Response, and (b) even someone with experience on the project would forget the distinction and need to inspect the spec to figure it out.

Reply:
Mostly to keep the DTOs names clean. We will remove the number-based types with (I think) the October '25 release. So, with 8.8, we'll only have the new string-based DTOs. There is nothing "special" about them anymore then, they have keys and those are strings.
Changing the existing number-based DTO names also doesn't work well since this would be a regression from 8.6 to 8.7 in the spec which is at least not great. The impact could be OK-ish but is not worth the issues it might cause.

Hope that makes sense in any way 😅